src/channel.cr

```diff
@@ -157,6 +157,14 @@ class Channel(T)
     receive_impl { return nil }
   end
 
+  # Receives a value from the channel.
+  # If there is a value waiting, it is returned immediately. Otherwise, this method returns nil.
+  #
+  # Returns `nil` if the channel is closed.
+  def nb_receive
+    @lock.sync { closed? ? nil : receive_internal { nil } }
```

Comment:
The closed check needs to move inside `receive_internal` like in `receive_impl`. And the behaviour of returning values after the channel is closed needs to have a spec (see https://github.com/crystal-lang/crystal/commit/3a9fa4ebd9922036d37f47c61ccee73a59d6d95e).

Reply:
I agree that it should be named differently (maybe `nreceive`? like `index` and `rindex`) and I like idea about add new behaviour to existed method but it requires more complex changes and really should be discussed.